UnitTests/tInductionSchemeGenerator.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+
+#include "Test/TestUtils.hpp"
+#include "Test/UnitTesting.hpp"
+#include "Test/SyntaxSugar.hpp"
+
+#include "Shell/InductionSchemeGenerator.hpp"
+
+using namespace Shell;
+
+#define MY_SYNTAX_SUGAR                                                                    \
+  DECL_DEFAULT_VARS                                                                        \
+  DECL_SORT(s)                                                                             \
+  DECL_SKOLEM_CONST(sk1,s)                                                                 \
+  DECL_SKOLEM_CONST(sk2,s)                                                                 \
+  DECL_SKOLEM_CONST(sk3,s)                                                                 \
+  DECL_CONST(b, s)                                                                         \
+  DECL_FUNC(r, {s}, s)                                                                     \
+  DECL_TERM_ALGEBRA(s, {b, r})                                                             \
+  DECL_FUNC(f, {s, s}, s)                                                                  \
+  DECL_FUNC(g, {s, s}, s)                                                                  \
+  DECL_PRED(p, {s, s})
+
+void checkResult(vvector<pair<InductionScheme, OccurrenceMap>> res,
```

Comment:
Maybe better name this "checkOccurrenceMap" or something like that, since it checks only the occ. map and not the scheme?